apps/web-app/components/convos/convoListItem.vue

```diff
@@ -11,128 +12,103 @@
   type Props = {
     convo: UserConvosDataType[number];
   };
-
   const props = defineProps<Props>();
 
   const timeAgo = useTimeAgo(props.convo.lastUpdatedAt || new Date());
-  const authorName = computed(() => {
-    if (
-      props.convo.messages[0].author.userProfile?.firstName &&
-      props.convo.messages[0].author.userProfile?.lastName
-    ) {
-      return (
-        props.convo.messages[0].author.userProfile.firstName +
-        ' ' +
-        props.convo.messages[0].author.userProfile.lastName
-      );
-    }
-    if (props.convo.messages[0].author.userGroup?.name) {
-      return props.convo.messages[0].author.userGroup.name + ' Team';
-    }
 
-    if (props.convo.messages[0].author.foreignEmailIdentity?.senderName) {
-      return props.convo.messages[0].author.foreignEmailIdentity.senderName;
-    }
-    return '';
+  const participantArray = ref<ConvoParticipantEntry[]>([]);
+  const author = ref<ConvoParticipantEntry>();
+  const authorEntryPublicId = computed(() => {
+    const entryAuthor = props.convo.entries[0].author;
+    return (
+      entryAuthor.orgMember?.publicId ||
+      entryAuthor.userGroup?.publicId ||
+      entryAuthor.contact?.publicId
+    );
   });
-
-  const authorAvatar = computed(() => {
-    if (props.convo.messages[0].author.userProfile?.avatarId) {
-      return props.convo.messages[0].author.userProfile.avatarId;
-    }
-    if (props.convo.messages[0].author.userGroup?.avatarId) {
-      return props.convo.messages[0].author.userGroup.avatarId;
-    }
-
-    if (props.convo.messages[0].author.foreignEmailIdentity?.avatarId) {
-      return props.convo.messages[0].author.foreignEmailIdentity.avatarId;
-    }
-    return '';
+  const authorEntryName = computed(() => {
+    const entryAuthor = props.convo.entries[0].author;
```

Comment:
is there any chance that entries is empty? If so this would error

Reply:
We should enable https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess

Reply:
good idea, add it project wide?